aptos-node/src/lib.rs

```diff
@@ -318,6 +316,57 @@ where
     start(config, Some(log_file), false)
 }
 
+/// Creates a simple test environment and starts the node.
+///
+/// You will notice many args referring to configs. Let's explain them:
+/// - `test_config_override_path` is the path to a config file that will be used as
+///   a template when building the final config. If not provided, a default template
+///   will be used. Many overrides are applied on top of this base config.
+/// - `config_path` is similar to `test_config_override_path`, but many of the
+///   overrides that are applied when using `test_config_override_path` are not
+///   applied when using `config_path`. Read the code for more info.
+/// - `config` is a complete NodeConfig. No overrides are applied on top of this if
+///    it is provided. If both `config` and `test_dir` are provided, `config` takes
+///    precedence.
+/// - `test_dir` is a directory that contains a config file. Much like `config`, the
+///   config read from this file is used without any overrides.
```

Comment:
Trying wrap my head around this...

Not sure if I read the code right, but If both `config_path` and `test_dir` are provided, the config from `test_dir` is ignored? Should we use the one from `test_dir` if provided?

Reply:
Actually the other way around, if `test_dir` is provided then `config_path` is ignored. This is a pretty awful function, I'm just trying to document it, an enum would be better here.

Reply:
I see! Yeah probably a good `ops week` item in Q4. Lets get this fix and do a release soon to unblock 😃 

nit: Intuitively, I feel like if `test_dir` is provided, I would assume `test_dir` is used and overrides everything else, including `config`

Reply:
Yeah I went back and forth on that. In reality this is just not good code and Rust has enums to solve this problem.